src/dbxdeploy/repos/RepoManager.py

```diff
@@ -48,12 +62,18 @@ def create_or_update(self, repo_url, branch, tag, repo_name, force):
                 self.__logger.info("Error occurred, cleaning up ...")
                 self.__repos_api.delete(repo_id)
             raise e
+        else:
+            self.__logger.info("Repo successfully updated")
 
-        self.__logger.info("Repo successfully updated")
-
-    def delete(self, repo_url, repo_name):
-        repo_path = self.__repo_path.format(repo_name=repo_name)
+    def delete(self, repo_url, repo_name, branch, tag):
+        if repo_name is None and "{repo_name}" in self.__repo_base_dir:
+            raise Exception("{repo_name} used in repo.base_dir, but --repo_name argument was not provided")
+        if branch is None and "{branch_name}" in self.__repo_base_dir:
+            raise Exception("{branch_name} used in repo.base_dir, but --branch argument was not provided")
+        if tag is None and "{tag_name}" in self.__repo_base_dir:
+            raise Exception("{tag_name} used in repo.base_dir, but --tag argument was not provided")
 
+        repo_path = self.__repo_base_dir.format(repo_name=repo_name, branch_name=branch, tag_name=tag)
```

Comment:
Let's create __resolve_repo_path() method which handles validation and format self.__repo_base_dir